### PR TITLE
fix: optional dependency issue

### PR DESCRIPTION
**Pull request by zahid-syed**

### CODE REVIEW

File: semantic_router/encoders/mistral.py [hunk 1]
```diff
@@ -26,33 +26,58 @@ def __init__(
...
         if api_key is None:
             raise ValueError("Mistral API key not provided")
         try:
-            self.client = MistralClient(api_key=api_key)
+            client = MistralClient(api_key=api_key)
+            embedding_response = EmbeddingResponse
+            mistral_exception = MistralException
         except Exception as e:
             raise ValueError(f"Unable to connect to MistralAI {e.args}: {e}") from e
+        return client, embedding_response, mistral_exception
```
Comment (ashraq1455) [inline semantic_router/encoders/mistral.py:56]:
```suggestion
        return client, mistralai
```

File: semantic_router/encoders/mistral.py [hunk 2]
```diff
@@ -26,33 +26,58 @@ def __init__(
...
 
         # Exponential backoff
         for _ in range(3):
             try:
-                embeds = self.client.embeddings(model=self.name, input=docs)
+                embeds = self._client.embeddings(model=self.name, input=docs)
                 if embeds.data:
                     break
-            except MistralException as e:
+            except self._mistral_exception as e:
```
Comment (ashraq1455) [inline semantic_router/encoders/mistral.py:70]:
```suggestion
            except self._mistralai.exceptions.MistralException as e:
```

File: semantic_router/encoders/mistral.py [hunk 3]
```diff
@@ -26,33 +26,58 @@ def __init__(
...
                 sleep(2**_)
                 error_message = str(e)
             except Exception as e:
                 raise ValueError(f"Unable to connect to MistralAI {e.args}: {e}") from e
 
-        if not embeds or not isinstance(embeds, EmbeddingResponse) or not embeds.data:
+        if (
+            not embeds
+            or not isinstance(embeds, self._embedding_response)
```
Comment (ashraq1455) [inline semantic_router/encoders/mistral.py:78]:
```suggestion
            or not isinstance(embeds, self._mistralai.models.embeddings.EmbeddingResponse)
```

**Review by ashraq1455 — CHANGES_REQUESTED** [review]

**Review by ashraq1455 — APPROVED** [review]
Lg

**Review by jamescalam — APPROVED** [review]
lgtm @zahid-syed, great work

**Pull request merged**